### PR TITLE
Various fixes

### LINKED ISSUE #11: Remove bulk_pull_blocks
`bulk_pull_blocks` was removed in https://github.com/nanocurrency/nano-node/pull/1615 in v18, can probably be removed from here now.

### LINKED ISSUE #18: Sequence has been replaced by timestamp
`vote_common` needs a rename in the spec

### LINKED ISSUE #23: Since UDP is now deprecated this comment should be reviewed.
https://github.com/nanocurrency/protocol/blob/e48566f2e02341b673a3f075e0e9b3ed25eedd85/reference/protocol.ksy#L266

### PR DESCRIPTION
**Pull request by zhyatt**
- Resolves telemetry field order Issue noted in https://github.com/nanocurrency/nano-node/issues/3490
- Resolves #11 
- Resolves #18 - I am not confident on the proper update being included here, but making a guess to move it along (type may be more appropriate possibly)
- Resolves #23 

**NOTE:** If any changes need to be made the `nano.png` file will need to be regenerated and committed

### CODE REVIEW

File: reference/protocol.ksy [hunk 1]
```diff
@@ -55,11 +54,10 @@ enums:
     0x06: bulk_pull
     0x07: bulk_push
     0x08: frontier_req
-    0x09: bulk_pull_blocks
-    0x0a: node_id_handshake
-    0x0b: bulk_pull_account
-    0x0c: telemetry_req
-    0x0d: telemetry_ack
+    0x09: node_id_handshake
```
Comment (dsiganos) [inline reference/protocol.ksy:57]: node_id_handshake must remain unchanged at value 0xA.
Same thing applies for the others below.

File: reference/protocol.ksy [hunk 2]
```diff
@@ -300,6 +298,8 @@ types:
         size: 64
       - id: sequence
         type: u8le
+      - id: timestamp
```
Comment (dsiganos) [inline reference/protocol.ksy:301]: timestamp replaces sequence, it is not a sibling of it

File: reference/protocol.ksy (hunk 1 above)
Comment (zhyatt) [reply reference/protocol.ksy]: Thanks, resolved!

File: reference/protocol.ksy (hunk 2 above)
Comment (zhyatt) [reply reference/protocol.ksy:299]: Ah, I was confusing it with other areas where we left them as siblings then. Thanks, resolved!

Comment (dsiganos) [reply reference/protocol.ksy:299]: The type does change either. It remains u8le.

Comment (zhyatt) [reply reference/protocol.ksy:299]: Ah, thanks...resolved!

**Comment by cryptocode** [comment]
According to the node source code, the node accepts telemetry_ack's that are larger than the current one, probably to be lenient during protocol upgrades.

This means that the protocol spec must ignore trailing data, just like the node code, otherwise generated parsers will get out of whack. This can be done by comparing current position to the header's `telemetry_size`. 

Adding the following at the very end of `msg_telemetry_ack:` should do the trick:

```
      - id: unknown_data
        type: u8
        repeat: until
        repeat-until: _io.pos == _root.header.telemetry_size
        if: _io.pos < _root.header.telemetry_size
```

**Comment by dsiganos** [comment]
@cryptocode You are correct that a telemetry_ack can contain some data at the end of it and that we need something extra to handle that case but I do not know if your proposed solution would work. I do not understand kaitai sufficiently well to say authoritative if your solution would work or not.

I would expect the solution to be using the telemetry_size field.
https://github.com/nanocurrency/protocol/blob/b94bb467bf5264603b8a1870524791279988783e/reference/protocol.ksy#L133

The size of any telemetry_ack is determined by the telemetry_size field, which is set in the extensions bits in the header.

**Comment by zhyatt** [comment]
@dsiganos @cryptocode This fell off my radar but I just updated per https://github.com/nanocurrency/protocol/pull/24#issuecomment-936109704

**Review by cryptocode — APPROVED** [review]

**Review by dsiganos — APPROVED** [review]

**Comment by dsiganos** [comment]
This looks good but we also need to document that the 4 LSB bits of the timestamp are now a duration field.

**Comment by zhyatt** [comment]
> This looks good but we also need to document that the 4 LSB bits of the timestamp are now a duration field.

I added a ticket for this here: https://github.com/nanocurrency/protocol/issues/26

**Pull request merged**